sources/packages/web/src/router/index.ts

```diff
@@ -18,12 +25,21 @@ const routes: Array<RouteRecordRaw> = [
     path: "/UserAsFunc",
     name: "UserAsFunc",
     component: UserAsFunc
-  },
+  }
 ];
 
 const router = createRouter({
   history: createWebHistory(process.env.BASE_URL),
   routes
 });
 
+router.beforeEach(async () => {
+  // Check Config
+  if (!AppConfigService.shared.isConfigReady()) {
+    console.log("Router: Setting up config and auth");
+    await AppConfigService.shared.init();
```

Comment:
Should it not happen at an earlier moment like "main.ts"?

Reply:
@andrewsignori-aot is the the bootstrap code in main.ts **asynchronous** ?